services/src/main/scala/cromwell/services/healthmonitor/HealthMonitorServiceActor.scala

```diff
@@ -0,0 +1,119 @@
+package cromwell.services.healthmonitor
+
+import java.util.concurrent.TimeoutException
+
+import akka.actor.{Actor, Scheduler, Timers}
+import akka.pattern.{after, pipe}
+import cromwell.util.GracefulShutdownHelper.ShutdownCommand
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+import scala.language.postfixOps
+import HealthMonitorServiceActor._
+import com.typesafe.config.Config
+import cromwell.services.ServiceRegistryActor.ServiceRegistryMessage
+import net.ceedubs.ficus.Ficus._
+
```

Comment:
This was massively inspired by the health monitor in rawls. My original goal was to generify it so that the checks were pluggable and then contribute this to workbench-commons. It turned out that the last mile wiring was going to be irritating so i gave up on that part.

That said, the spirit of how this works is basically the same as theirs.